src/qibo/gates/gates.py

```diff
@@ -864,6 +864,27 @@ def _dagger(self) -> "Gate":
         theta, lam, phi = tuple(-x for x in self.parameters)  # pylint: disable=E1130
         return self.__class__(self.target_qubits[0], theta, phi, lam)
 
+    def decompose(self) -> List[Gate]:
+        """A global phase difference exists between the definitions of
+        :math:`U3` and this decomposition. More precisely,
```

Comment:
Just a small note on the docstring style: the first line should not wrap, and just be a very brief summary of the function (e.g. `pydocstyle` would check for the first line ending with a `.`).

Everything else can be as long as you wish.